dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/basicToolApp/BasicTools.java

```diff
@@ -150,4 +151,26 @@ public boolean toggle(@ToolArg(name = "value", description = "boolean value") bo
         return !value;
     }
 
+    /////////
+    //Annotations
+    /////////
+
+    @Tool(name = "readOnlyTool", title = "Read Only Tool", description = "A tool that is read-only",
+          annotations = @Annotations(readOnlyHint = true, destructiveHint = true, idempotentHint = false, openWorldHint = true, title = "Read Only Tool"))
+    public String readOnlyTool(@ToolArg(name = "input", description = "input string") String input) {
+        return input;
+    }
+
+    @Tool(name = "destructiveTool", title = "Destructive Tool", description = "A tool that performs a destructive operation",
+          annotations = @Annotations(readOnlyHint = false, destructiveHint = true, idempotentHint = false, openWorldHint = false, title = "Destructive Tool"))
+    public String destructiveTool(@ToolArg(name = "input", description = "input string") String input) {
+        return input;
+    }
+
+    @Tool(name = "openWorldTool", title = "Open to World Tool", description = "A tool in an open world context",
+          annotations = @Annotations(readOnlyHint = false, destructiveHint = false, idempotentHint = false, openWorldHint = true, title = "Open to World Tool"))
+    public String openWorldTool(@ToolArg(name = "input", description = "input string") String input) {
+        return input;
+    }
```

Comment:
Can you add a test case where you don't provide all of the `@Annotations` parameters please?

E.g. `@Anntotations(idempotentHint = true, destructiveHint = false)`